usbmanager golang panic: missing checks for nil return from ioBundle2PassthroughRule?

Please provide information requested in the template below as much as possible.

#### Describe the bug

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x230f775]

goroutine 291 [running]:
github.com/lf-edge/eve/pkg/pillar/cmd/usbmanager.(*usbmanagerController).addVirtualmachine(0xc001135b98, {{0xc001bfddc0, 0x40}, {0xc001d84180, 0x1, 0x1}})
	/pillar/cmd/usbmanager/usbcontroller.go:129 +0x375
github.com/lf-edge/eve/pkg/pillar/cmd/usbmanager.(*usbmanagerContext).handleDomainStatusRunning(_, {{{0x69, 0xf6, 0x9a, 0x11, 0x3b, 0xb6, 0x4b, 0x7d, 0x96, ...}, ...}, ...})
	/pillar/cmd/usbmanager/subscriptions.go:124 +0x2e8
github.com/lf-edge/eve/pkg/pillar/cmd/usbmanager.(*usbmanagerContext).handleDomainStatusModify(0xc001135b20, {0x29c9680?, 0xc0004c3b80?}, {0xfe6250?, 0xc0019b8180?}, {0x29c9680?, 0xc001d8a280?}, {0x24?, 0x29c9680?})
	/pillar/cmd/usbmanager/subscriptions.go:98 +0x165
github.com/lf-edge/eve/pkg/pillar/pubsub.handleModify({0x29331c0?, 0xc001101400}, {0xc001d6e2a0, 0x24}, {0xc001d7e000, 0x5a2, 0x5a3})
	/pillar/pubsub/subscribe.go:273 +0x6ea
github.com/lf-edge/eve/pkg/pillar/pubsub.(*SubscriptionImpl).ProcessChange(0xc001101400, {0x3, {0xc001d6e2a0, 0x24}, {0xc001d7e000, 0x5a2, 0x5a3}})
	/pillar/pubsub/subscribe.go:133 +0x3d7
github.com/lf-edge/eve/pkg/pillar/cmd/usbmanager.(*usbmanagerContext).process.func1({0x28035c0?, 0xc000a3a600?})
	/pillar/cmd/usbmanager/subscriptions.go:27 +0xca
github.com/lf-edge/eve/pkg/pillar/pubsub.MultiChannelWatch({0xc001053580, 0x3, 0x0?})
	/pillar/pubsub/util.go:133 +0x25c
github.com/lf-edge/eve/pkg/pillar/cmd/usbmanager.(*usbmanagerContext).process(0xc001135b20, 0xc0012e9ef0)
	/pillar/cmd/usbmanager/subscriptions.go:39 +0x42f
github.com/lf-edge/eve/pkg/pillar/cmd/usbmanager.Run(0xc0012e9ef0, 0x0?, 0x0?, {0x441fa18, 0x0, 0x0})
	/pillar/cmd/usbmanager/usbmanager.go:93 +0x3ee
main.startAgentAndDone({0x2af0a50?, 0x0?}, {0xc0012d5240, 0xa}, 0x0?, 0x0?, 0x0?, {0x441fa18, 0x0, 0x0})
	/pillar/zedbox/zedbox.go:238 +0x68
created by main.handleService
	/pillar/zedbox/zedbox.go:228 +0x38a

#### Recent release

master as of today

#### Duplicate

If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
pointing to the existing issue instead.

#### How to reproduce

Include a list of steps showing how to reproduce the bug. Also include:

* EVE-OS version [e.g., 6.6.3]
* EVE-OS hardware architecture [e.g., arm64, amd64]

#### Expected behavior

A clear and concise description of what you expected to happen.

#### Error message

If applicable, add screenshots to help explain your problem.

#### Additional context

Add any other information about the problem here.

--- a/pkg/pillar/cmd/usbmanager/usbcontroller.go
+++ b/pkg/pillar/cmd/usbmanager/usbcontroller.go
@@ -236,7 +236,8 @@ func ioBundle2PassthroughRule(adapter types.IoBundle) passthroughRule {
 		}
 		busnum, err := strconv.ParseUint(usbParts[0], 10, 16)
 		if err != nil {
-			panic(err)
+			log.Warnf("usbaddr busnum (%s) not parseable", usbParts[0])
+			return nil
 		}
 		portnum := usbParts[1]
 		ud := usbdevice{

--- a/pkg/pillar/cmd/usbmanager/usbcontroller.go
+++ b/pkg/pillar/cmd/usbmanager/usbcontroller.go
@@ -126,6 +126,9 @@ func (uc *usbmanagerController) addVirtualmachine(vm virtualmachine) {
 		}
 
 		pr := ioBundle2PassthroughRule(*ioBundle)
+		if pr == nil {
+			continue
+		}
 		pr.setVirtualMachine(&vm)
 
 		uc.ruleEngine.addRule(pr)
@@ -159,6 +162,9 @@ func (uc *usbmanagerController) removeVirtualmachine(vm virtualmachine) {
 		}
 
 		pr := ioBundle2PassthroughRule(*ioBundle)
+		if pr == nil {
+			continue
+		}
 		uc.ruleEngine.delRule(pr)
 	}
 
@@ -172,6 +178,9 @@ func (uc *usbmanagerController) removeIOBundle(ioBundle types.IoBundle) {
 	uc.usbpassthroughs.delIoBundle(&ioBundle)
 
 	pr := ioBundle2PassthroughRule(ioBundle)
+	if pr == nil {
+		return
+	}
 	vm := uc.usbpassthroughs.vmsByIoBundlePhyLabel[ioBundle.Phylabel]
 	pr.setVirtualMachine(vm)
 
@@ -197,7 +206,6 @@ func (uc *usbmanagerController) addIOBundle(ioBundle types.IoBundle) {
 
 	pr := ioBundle2PassthroughRule(ioBundle)
 	if pr == nil {
-		log.Tracef("unusable iobundle %s: %s/%s\n", ioBundle.Phylabel, ioBundle.UsbAddr, ioBundle.PciLong)
 		return
 	}
 	vm := uc.usbpassthroughs.vmsByIoBundlePhyLabel[ioBundle.Phylabel]
@@ -249,6 +257,7 @@ func ioBundle2PassthroughRule(adapter types.IoBundle) passthroughRule {
 
 		pr = &usb
 	} else {
+		log.Tracef("cannot create rule out of adapter %+v\n", adapter)
 		pr = nil
 	}
 

--- a/pkg/pillar/cmd/usbmanager/usbcontroller_test.go
+++ b/pkg/pillar/cmd/usbmanager/usbcontroller_test.go
@@ -71,6 +71,27 @@ func testEventTable() [][]testingEvent {
 	return tetg.testEventTable
 }
 
+func TestAddNonRuleIOBundle(t *testing.T) {
+	uc := newTestUsbmanagerController()
+	ioBundle := types.IoBundle{
+		Type:         0,
+		Phylabel:     "Test",
+		Logicallabel: "Test",
+	}
+	uc.addIOBundle(ioBundle)
+	uc.removeIOBundle(ioBundle)
+
+	uc.usbpassthroughs.ioBundles[ioBundle.Phylabel] = &ioBundle
+
+	vm := virtualmachine{
+		qmpSocketPath: "",
+		adapters:      []string{"Test"},
+	}
+
+	uc.addVirtualmachine(vm)
+	uc.removeVirtualmachine(vm)
+}
+
 func TestRemovingVm(t *testing.T) {
 	usbEventBusnum := uint16(1)
 	usbEventDevnum := uint16(2)

